examples/tutorials/date_time_charts.py

```diff
@@ -0,0 +1,351 @@
+"""
+Plotting datetime charts
+========================
+
+PyGMT accepts a variety of datetime objects to plot data and create charts.
+Aside from the built-in Python datetime object, PyGMT supports input using
+properly formatted strings, ``pandas``, ``xarray``, as well as ``numpy``.
+These data types can be used to plot specific points as well as get
+passed into the ``region`` parameter to create a range of the data on an axis.
+
+The following examples will demonstrate how to create plots
+using the different datetime objects.
+"""
+# sphinx_gallery_thumbnail_number = 0
+
+import datetime
+
+import numpy as np
+import pandas as pd
+import pygmt
+import xarray as xr
+
+###############################################################################
+# Using Python's ``datetime``
+# ---------------------------
+#
+# In this example, Python's built-in ``datetime`` module is used
+# to create data points stored in list ``x``. Additionally,
+# dates are passed into the ``region`` parameter in the format
+# ``(x_start, x_end, y_start, y_end)``,
+# where the date range is plotted on the x-axis.
+# An additional notable parameter is ``style``, where it's specified
+# that data points are to be plotted in an **X** shape with a size
+# of 0.3 centimeters.
+#
+
+x = [
+    datetime.date(2010, 6, 1),
+    datetime.date(2011, 6, 1),
+    datetime.date(2012, 6, 1),
+    datetime.date(2013, 6, 1),
+]
+y = [1, 2, 3, 5]
+
+fig = pygmt.Figure()
+fig.plot(
+    projection="X10c/5c",
+    region=[datetime.date(2010, 1, 1), datetime.date(2014, 12, 1), 0, 6],
+    frame=["WSen", "afg"],
+    x=x,
+    y=y,
+    style="x0.3c",
+    pen="1p",
+)
+fig.show()
+
+###############################################################################
+# In addition to specifying the date, ``datetime`` supports
+# the exact time at which the data points were recorded. Using :meth:`datetime.datetime`
+# the ``region`` parameter as well as data points can be created
+# with both date and time information.
+#
+# Some notable differences to the previous example include
+#
+# - Modifying ``frame`` to only include West(left) and South
+# (bottom) border, and removing grid lines
+# - Using circles to plot data points defined through ``c``
+# in ``style`` parameter
```

Comment:
It seems that the ordered list does not work.

![list](https://user-images.githubusercontent.com/50591376/120162290-d2717d00-c22a-11eb-9c61-29cddf1f48a9.png)